lib/pbench/server/cache_manager.py

```diff
@@ -430,29 +429,61 @@ def get_info(self, path: Path) -> JSONOBJECT:
         return fd_info
 
     @staticmethod
-    def extract(tarball_path: Path, path: str) -> Optional[str]:
-        """Extract a file from the tarball and return it as a string
+    def filestream(tarball_path: Path, path: Path) -> Optional[IO[bytes]]:
+        """Extracts filestream from the tarball
```

Comment:
I think we could use a better description.  This function doesn't really "extract" anything -- it returns a file stream which yields the contents of the file at the specified path in the specified tarball...if the caller reads it.  🙂